jina/peapods/peas/__init__.py

```diff
@@ -383,94 +262,47 @@ def close_zmqlet(self):
 
     def _initialize_executor(self):
         try:
-            self.set_environment_vars()
             self.load_plugins()
             self.load_executor()
             return self.executor
-        except ExecutorFailToLoad:
+        except Exception as ex:
             self.logger.critical(f'can not start a executor from {self.args.uses}', exc_info=True)
+            raise ex
 
-    def run(self):
+    def run(self, is_ready_event: 'Event'):
```

Comment:
Rather than passing `is_ready_event` to different `run` methods, instantiating Pea with the event (in `__init__`) might be more elegant?

Reply:
it may be more elegant but I do want to be clear about ownership

Reply:
and that is communicating to the outside world